### PR TITLE
OtTracerPropagator: send only 64 bits for trace id

### PR DESCRIPTION
**Pull request by malafeev**
cc @carlosalberto

### CODE REVIEW

File: extensions/trace-propagators/src/test/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagatorTest.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
   private static final TraceState TRACE_STATE_DEFAULT = TraceState.builder().build();
   private static final String TRACE_ID = "ff000000000000000000000000000041";
+  private static final String TRACE_ID_RIGHT_PART = TRACE_ID.substring(TraceId.getHexLength() / 2);
```
Comment (carlosalberto) [inline extensions/trace-propagators/src/test/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagatorTest.java:30]: In order to have this in a more explicit fashion, I'd simply go and use the hardcoded value itself, as other values do here, i.e. `TRACE_ID_RIGHT_PART = "0000000000000041"`.

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java [hunk 2]
```diff
@@ -58,7 +58,10 @@ public static OtTracerPropagator getInstance() {
     if (!spanContext.isValid()) {
       return;
     }
-    setter.set(carrier, TRACE_ID_HEADER, spanContext.getTraceIdAsHexString());
+    setter.set(
```
Comment (carlosalberto) [inline extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java:61]: Leave a comment explaining we are doing this explicitly as the format only supports 64-bit ids?

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Looks good! Left a pair of comments but overall LGTM

**Review by anuraaga — COMMENTED** [review]
Do we need a custom id generator too?

File: extensions/trace-propagators/src/test/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagatorTest.java (hunk 1 above)
Comment (malafeev) [reply extensions/trace-propagators/src/test/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagatorTest.java]: done

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java (hunk 2 above)
Comment (malafeev) [reply extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java:63]: done

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java [hunk 3]
```diff
@@ -58,7 +58,12 @@ public static OtTracerPropagator getInstance() {
     if (!spanContext.isValid()) {
       return;
     }
-    setter.set(carrier, TRACE_ID_HEADER, spanContext.getTraceIdAsHexString());
+    // OpenTracing trace id MUST be 64-bits therefore OpenTelemetry trace id is truncated to 64-bits
```
Comment (anuraaga) [inline extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java:61]:
```suggestion
    // Lightstep trace id MUST be 64-bits therefore OpenTelemetry trace id is truncated to 64-bits
```

This is specific to Lightstep I guess? Opentracing spec says 128-bit

https://github.com/opentracing/specification/blob/master/rfc/trace_identifiers.md#trace-context-http-headers

**Comment by malafeev** [comment]
> Do we need a custom id generator too?

I don't think we need custom id generator. cc @carlosalberto

**Comment by anuraaga** [comment]
> I don't think we need custom id generator. cc @carlosalberto

I'm guessing this is because collector, or whatever other receiver, will also truncate IDs it receives, so they match up. Makes sense to me

**Review by anuraaga — APPROVED** [review]

**Comment by carlosalberto** [comment]
>  will also truncate IDs it receives

Yes, this is the case :)

**Comment by malafeev** [comment]
Looks like PR is ready to merge, all comments resolved. Can anybody merge it?

**Comment by carlosalberto** [comment]
Merged (although not a maintainer anymore. Hope that is ok @jkwatson @anuraaga ;) )

**Pull request merged**